firefox-ios/Client/Experiments/Messaging/GleanPlumbMessage.swift

```diff
@@ -89,6 +89,12 @@ struct GleanPlumbMessage {
     var options: [String] {
         return data.microsurveyConfig?.options ?? []
     }
+
+    /// The icon for this message if it has a microsurvey configuration.
+    /// Embedding apps should not read from this directly.
+    var icon: UIImage? {
+        return data.microsurveyConfig?.icon
```

Comment:
@adudenamedruby all works well if the configuration has the correct values, but if the values are incorrect, we hit a `fatalError`. This is due to hitting `getImageNotNull` in the nimbus generated files. There is a separate method that we can use `getImage`, but since that file is generated I may need to reach out to nimbus team. Let me know if you have additional context that I may be missing, thanks!

![image](https://github.com/mozilla-mobile/firefox-ios/assets/6743397/a4100db9-1a60-4df0-8a93-3fb020dddc58)

Reply:
Discussed with Jared from Nimbus team on this, it seems like I'll need to make the change in the app services repo. I'll tag you in that PR as well, but let me know if you see any issues with this change.

https://github.com/mozilla/application-services/blob/71578bea70b5f9da0b9c35c17e8fe15e9cf2ebc9/components/support/nimbus-fml/src/backends/swift/gen_structs/bundled.rs#L157